## 1. Import Library serta Melakukan Eksplor Dataset Dengan Pandas

In [235]:
# Import library yang akan digunakan dalam melakukan prediksi
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
plt.style.use("seaborn")
import warnings
warnings.filterwarnings("ignore")
from sklearn.metrics import mean_absolute_error, mean_squared_error

In [236]:
# Read dataset
data_februari = pd.read_csv("volume-pengangkutan-sampah-di-kali-sungai-situ-waduk-bulan-februari-tahun-2021.csv", 
                            encoding="latin-1")
data_februari.tail()

bulan                                      titik_lokasi   kecamatan   \
49247      2  Kali Cakung Drain (Jembatan Tinggi-Perbatasan ...      Cakung   
49248      2                                   PHB Rawa Teratai      Cakung   
49249      2                                 PHB Rawa Gelam III      Cakung   
49250      2                                   PHB Metropos KIP      Cakung   
49251      2                                 PHB Warung Jengkol      Cakung   

            wilayah   panjang/luas  satuan_panjang/luas  tanggal  \
49247  Jakarta Timur            800                   m       28   
49248  Jakarta Timur            700                   m       28   
49249  Jakarta Timur           1000                   m       28   
49250  Jakarta Timur           1000                   m       28   
49251  Jakarta Timur            400                   m       28   

      volume_sampah_perhari(m3)  
49247                         -  
49248                         -  
49249                         -  
49250                         -  
49251                         -

In [237]:
# Melihat info dataset (tipe data, total record data, dll)
data_februari.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 49252 entries, 0 to 49251
Data columns (total 8 columns):
 #   Column                     Non-Null Count  Dtype 
---  ------                     --------------  ----- 
 0   bulan                      49252 non-null  int64 
 1    titik_lokasi              49252 non-null  object
 2    kecamatan                 49252 non-null  object
 3    wilayah                   49252 non-null  object
 4    panjang/luas              39816 non-null  object
 5   satuan_panjang/luas        29428 non-null  object
 6   tanggal                    49252 non-null  int64 
 7   volume_sampah_perhari(m3)  49252 non-null  object
dtypes: int64(2), object(6)
memory usage: 3.0+ MB


In [238]:
# Menghapus jarak spasi pada kolom dataset
data_februari.columns = data_februari.columns.str.replace(" ", "")

In [239]:
# Melihat nilai statistik deskriptif dari dataframe untuk keseluruhan kolom
data_februari.describe(include="all")

bulan titik_lokasi  kecamatan        wilayah panjang/luas  \
count   49252.0        49252      49252          49252        39816   
unique      NaN         1753         42              5          299   
top         NaN  PHB Pramuka  Kalideres  Jakarta Barat         1000   
freq        NaN           56       2632          14840         5936   
mean        2.0          NaN        NaN            NaN          NaN   
std         0.0          NaN        NaN            NaN          NaN   
min         2.0          NaN        NaN            NaN          NaN   
25%         2.0          NaN        NaN            NaN          NaN   
50%         2.0          NaN        NaN            NaN          NaN   
75%         2.0          NaN        NaN            NaN          NaN   
max         2.0          NaN        NaN            NaN          NaN   

       satuan_panjang/luas       tanggal volume_sampah_perhari(m3)  
count                29428  49252.000000                     49252  
unique                   9           NaN                        93  
top                      m           NaN                         0  
freq                 26964           NaN                     14029  
mean                   NaN     14.500000                       NaN  
std                    NaN      8.077829                       NaN  
min                    NaN      1.000000                       NaN  
25%                    NaN      7.750000                       NaN  
50%                    NaN     14.500000                       NaN  
75%                    NaN     21.250000                       NaN  
max                    NaN     28.000000                       NaN

Hasil keluaran diatas menandakan beberapa nilai-nilai dari statistik deskriptif dari kedua jenis data yakni numerik (tipe data int & float) dan kategorik (tipe data object/string) dan tentu saja setiap jenis data memiliki nilai deskriptif yang berbeda. Seperti misalnya jenis data numerik memiliki nilai deskriptif seperti count, mean, std (standard deviasi), maximum, minimum, quartil 1 (25%), quartil 2/median (50%) dan quartil 3 (75%). Sedangkan jenis data kategorik memiliki nilai deskriptif seperti count, unique, top dan freq (frekuensi data).

Sebagai contoh dapat dilihat pada penjelasan dibawah ini:

- Pada kolom titik_lokasi tidak memiliki nilai mean, standard deviasi, min, dll, tetapi memiliki nilai top, unique, freq, dll. Hal tersebut dikarenakan kolom titik_lokasi bersifat kategorik (karena memiliki tipe data string). Maka dari itu kolom titik_lokasi memiliki nilai top yakni PHB Pramuka, nilai freq adalah 56, dst.

- Sedangkan pada kolom tanggal tidak memiliki nilai top, freq, dll tetapi memiliki nilai mean, standard deviasi, min, max, dll. Hal tersebut dikarenakan kolom tanggal bersifat numerik. Hal ini dapat dilihat dari nilai mean pada kolom tanggal adalah 14.5, nilai max adalah 28.0, nilai quartil 3 adalah 21.25, dst.

Berikut adalah kolom-kolom yang bersifat kategorik dan numerik:

- Kategorik: titik_lokasi, kecamatan, wilayah, panjang/luas, satuan_panjang/luas dan volume_sampah_perhari(m3).
- Numerik: bulan dan tanggal.

In [240]:
# Melihat tipe data dari masing-masing kolom yang ada di dalam dataframe
data_februari.dtypes

bulan                         int64
titik_lokasi                 object
kecamatan                    object
wilayah                      object
panjang/luas                 object
satuan_panjang/luas          object
tanggal                       int64
volume_sampah_perhari(m3)    object
dtype: object

In [241]:
# Melihat nama kolom yang ada di dalam dataframe
data_februari.columns

Index(['bulan', 'titik_lokasi', 'kecamatan', 'wilayah', 'panjang/luas',
       'satuan_panjang/luas', 'tanggal', 'volume_sampah_perhari(m3)'],
      dtype='object')

In [242]:
# Melihat jumlah rows dan kolom pada dataframe
data_februari.shape

(49252, 8)

Sangat berbeda sekali jumlah rows atau jumlah baris dari dataframe bulan Februari dengan bulan Januari. Pada bulan Januari jumlah rows-nya sebanyak 54343 sedangkan jumlah rows di bulan Februari yakni mencapai 49252.

In [243]:
"""
Menghitung nilai dari beberapa kolom (kecamatan, wilayah, satuan_panjang/satuan) 
dari yang sering muncul - tidak sering muncul di dalam dataframe
"""
print(data_februari["kecamatan"].value_counts())
print()

print(data_februari["wilayah"].value_counts())
print()

print(data_februari["satuan_panjang/luas"].value_counts())

Kalideres            2632
Tanjung Priok        2632
Penjaringan          2492
Cilincing            2296
Cengkareng           2240
Grogol Petamburan    2100
Cakung               2044
Kembangan            2016
Tamansari            1764
Pademangan           1708
Kebon Jeruk          1428
Tambora              1400
Kelapa Gading        1344
Palmerah             1260
Koja                 1260
Jatinegara           1232
Duren Sawit          1176
Pulo Gadung          1176
Jagakarsa            1092
Pasar Rebo           1036
Kramat Jati          1008
Matraman              868
Makasar               868
Gambir                840
Ciracas               840
Tanah Abang           812
Pancoran              784
Cilandak              756
Pasar Minggu          756
Kebayoran Baru        756
Kebayoran Lama        728
Setiabudi             700
Tebet                 672
Mampang Prapatan      644
Cipayung              588
Pesanggrahan          560
Sawah Besar           504
Senen                 476
Kemayoran   

In [244]:
"""
Mengubah nama kolom satuan_panjang/luas menjadi satuan_panjang untuk melihat data yang disesuaikan. Seperti misalkan ingin
melihat data kecamatan yang hanya berisi informasi kecamatan Tanjung Priok, dst
"""
rename_column_data = data_februari.rename(columns={"satuan_panjang/luas": "satuan_panjang", 
                                                   "volume_sampah_perhari(m3)": "volume_sampah_perhari", 
                                                   "panjang/luas": "panjang_luas"})

rename_column_data.columns

Index(['bulan', 'titik_lokasi', 'kecamatan', 'wilayah', 'panjang_luas',
       'satuan_panjang', 'tanggal', 'volume_sampah_perhari'],
      dtype='object')

In [245]:
# Memisahkan semua data yang memiliki kecamatan = Tanjung Priok
data_tanjung_priok = rename_column_data[rename_column_data.kecamatan == "Tanjung Priok"]
data_tanjung_priok.head(10)

bulan                                  titik_lokasi      kecamatan  \
332      2  PHB Sunter Ready Mix Ruas bypass Yos Sudarso  Tanjung Priok   
333      2                 Kali Lagoa Tirem depan Sukmul  Tanjung Priok   
334      2               Kali Lagoa Tirem Depan Saringan  Tanjung Priok   
335      2                         Kali Lagoa Buntu Enim  Tanjung Priok   
336      2                      Kali Lagoa Buntu Arhanud  Tanjung Priok   
337      2             Kali Lagoa Kanal Jembatan Arhanud  Tanjung Priok   
338      2                      Kali Lagoa Kanal Enggano  Tanjung Priok   
339      2               Kali Lagoa Buntu Jembatan Bugis  Tanjung Priok   
340      2          Kali Sentiong Jembatan Jubile School  Tanjung Priok   
341      2                          Kali Sentiong Blok A  Tanjung Priok   

           wilayah panjang_luas satuan_panjang  tanggal volume_sampah_perhari  
332  Jakarta Utara          NaN            NaN        1                     2  
333  Jakarta Utara          NaN            NaN        1                     3  
334  Jakarta Utara          NaN            NaN        1                     0  
335  Jakarta Utara          NaN            NaN        1                     3  
336  Jakarta Utara          NaN            NaN        1                     2  
337  Jakarta Utara          NaN            NaN        1                     2  
338  Jakarta Utara          NaN            NaN        1                     3  
339  Jakarta Utara          NaN            NaN        1                     3  
340  Jakarta Utara          NaN            NaN        1                     1  
341  Jakarta Utara          NaN            NaN        1                     1

In [246]:
# Memisahkan semua data yang memiliki wilayah = Jakarta Pusat
data_jakpus = rename_column_data[rename_column_data.wilayah == "Jakarta Pusat"]
data_jakpus.head()

bulan                                 titik_lokasi   kecamatan  \
0      2       Kali / Sungai Sentiong - Ruas Dewi Sri  Johar Baru   
1      2         Kali / Sungai Sentiong - Ruas Abapon  Johar Baru   
2      2     Kali / Sungai Sentiong - JL Tanah Tinggi  Johar Baru   
3      2  Kali / Sungai Sentiong Jl.Galur - Pombensin  Johar Baru   
4      2           Kali / Sungai Sentiong - Jl. Timah  Johar Baru   

         wilayah panjang_luas satuan_panjang  tanggal volume_sampah_perhari  
0  Jakarta Pusat          464              m        1                     2  
1  Jakarta Pusat          464              m        1                     3  
2  Jakarta Pusat          464              m        1                     3  
3  Jakarta Pusat          606              m        1                     4  
4  Jakarta Pusat          310              m        1                     7

In [247]:
# Memisahkan semua data yang memiliki satuan_panjang = m (meter)
data_m = rename_column_data[rename_column_data.satuan_panjang == "m"]
data_m.head(10)

bulan                                 titik_lokasi   kecamatan  \
0      2       Kali / Sungai Sentiong - Ruas Dewi Sri  Johar Baru   
1      2         Kali / Sungai Sentiong - Ruas Abapon  Johar Baru   
2      2     Kali / Sungai Sentiong - JL Tanah Tinggi  Johar Baru   
3      2  Kali / Sungai Sentiong Jl.Galur - Pombensin  Johar Baru   
4      2           Kali / Sungai Sentiong - Jl. Timah  Johar Baru   
5      2                          PHB Kawi Kawi Bawah  Johar Baru   
6      2                                PHB Kayu Awet  Johar Baru   
7      2                              PHB Pulo Gundul  Johar Baru   
8      2                         PHB Tanah Tinggi XII  Johar Baru   
9      2                                PHB Bala Dewa  Johar Baru   

         wilayah panjang_luas satuan_panjang  tanggal volume_sampah_perhari  
0  Jakarta Pusat          464              m        1                     2  
1  Jakarta Pusat          464              m        1                     3  
2  Jakarta Pusat          464              m        1                     3  
3  Jakarta Pusat          606              m        1                     4  
4  Jakarta Pusat          310              m        1                     7  
5  Jakarta Pusat          620              m        1                   0.5  
6  Jakarta Pusat          300              m        1                    12  
7  Jakarta Pusat          500              m        1                     1  
8  Jakarta Pusat          500              m        1                     1  
9  Jakarta Pusat          330              m        1                     1

In [248]:
# Memisahkan semua data yang memiliki volume_sampah_perhari(m3) = - 
data_strip = rename_column_data[rename_column_data.volume_sampah_perhari == "-"]
data_strip.head(10)

bulan                                       titik_lokasi  kecamatan  \
55       2                                     PHB Abdul Muis     Gambir   
56       2                                         PHB Petojo     Gambir   
57       2                                  PHB Budi Kemulian     Gambir   
578      2  Kali Cengkareng Drain (Depan Pasar Kemiri Jl. ...  Kembangan   
581      2                    Kali PHB RW. 04 (Stasiun Pompa)  Kembangan   
590      2       Kali Pesanggrahan (Jln. H. Nabet - Metro TV)  Kembangan   
599      2                         Danau Hutan Kota Srengseng  Kembangan   
600      2                    PHB Jambu Rt 07 Rw 07 Srengseng  Kembangan   
602      2                         PHB Komplek BPPT Srengseng  Kembangan   
603      2                   PHB Pesanggrahan Jl. Meruya ilir  Kembangan   

           wilayah panjang_luas satuan_panjang  tanggal volume_sampah_perhari  
55   Jakarta Pusat         1000              m        1                     -  
56   Jakarta Pusat          500              m        1                     -  
57   Jakarta Pusat          500              m        1                     -  
578  Jakarta Barat         1500              m        1                     -  
581  Jakarta Barat         1000              m        1                     -  
590  Jakarta Barat         1500              m        1                     -  
599  Jakarta Barat          1.5             HA        1                     -  
600  Jakarta Barat          500              m        1                     -  
602  Jakarta Barat          600              m        1                     -  
603  Jakarta Barat          600              m        1                     -

In [249]:
data_panjangluas = rename_column_data[rename_column_data.panjang_luas == "3500/70"]
data_panjangluas.head()

bulan                          titik_lokasi  kecamatan        wilayah  \
157       2  Kali BKT Marunda Pintu Air Sekatan 1  Cilincing  Jakarta Utara   
1916      2  Kali BKT Marunda Pintu Air Sekatan 1  Cilincing  Jakarta Utara   
3675      2  Kali BKT Marunda Pintu Air Sekatan 1  Cilincing  Jakarta Utara   
5434      2  Kali BKT Marunda Pintu Air Sekatan 1  Cilincing  Jakarta Utara   
7193      2  Kali BKT Marunda Pintu Air Sekatan 1  Cilincing  Jakarta Utara   

     panjang_luas satuan_panjang  tanggal volume_sampah_perhari  
157       3500/70            NaN        1                     0  
1916      3500/70            NaN        2                     0  
3675      3500/70            NaN        3                    12  
5434      3500/70            NaN        4                     0  
7193      3500/70            NaN        5                    52

## 2. Data Preprocessing

In [250]:
# Mengecek apakah di dalam dataframe terdapat nilai duplikat/tidak
data_februari.duplicated().sum()

0

In [251]:
# Mengubah format penamaan kolom dari yang semula huruf kecil menjadi huruf gabungan (huruf besar & kecil)
data_februari = data_februari.rename(str.title, axis="columns")
data_februari.columns

Index(['Bulan', 'Titik_Lokasi', 'Kecamatan', 'Wilayah', 'Panjang/Luas',
       'Satuan_Panjang/Luas', 'Tanggal', 'Volume_Sampah_Perhari(M3)'],
      dtype='object')

In [252]:
# Mengecek nilai kosong (NaN) dari masing-masing kolom di dataframe
data_februari.isnull().sum()

Bulan                            0
Titik_Lokasi                     0
Kecamatan                        0
Wilayah                          0
Panjang/Luas                  9436
Satuan_Panjang/Luas          19824
Tanggal                          0
Volume_Sampah_Perhari(M3)        0
dtype: int64

Jika dilihat dari hasil diatas maka hanya terdapat 2 kolom yang memiliki nilai NaN (Not a Number) atau nilai kosong yaitu kolom "Panjang/Luas" dan "Satuan_Panjang/Luas". Namun jika diperhatikan kembali terdapat 3 kolom yang memiliki nilai NaN yakni kolom "Panjang/Luas", "Satuan_Panjang/Luas", dan kolom "Volume_Sampah_Perhari(M3)".

Mengapa terdapat kolom "Volume_Sampah_Perhari(M3)"?

Hal tersebut dikarenakan nilai kosong yang ada di kolom "Volume_Sampah_Perhari(M3)" ditulis dengan simbol "-". Karena kolom ini merupakan kolom prediksi maka dari itu saya akan mengubahnya menjadi nilai mean atau nilai rata-rata. Tetapi sebelum itu nilai "-" ini akan saya ubah terlebih dahulu menjadi nilai NaN (Not a Number).

In [253]:
# Menghapus karakter yang berada di kolom "Panjang/Luas" seperti pada data 3500/70 yang dapat dilihat pada tahapan sebelumnya
data_februari["Panjang/Luas"] = data_februari["Panjang/Luas"].str.replace(r"\D", "")

In [254]:
# Mengahapus karakter "-" di kolom "Volume_Sampah_Perhari(M3)" dengan nilai NaN
data_februari["Volume_Sampah_Perhari(M3)"] = data_februari["Volume_Sampah_Perhari(M3)"].replace("-", np.nan)

data_februari.tail()

Bulan                                       Titik_Lokasi Kecamatan  \
49247      2  Kali Cakung Drain (Jembatan Tinggi-Perbatasan ...    Cakung   
49248      2                                   PHB Rawa Teratai    Cakung   
49249      2                                 PHB Rawa Gelam III    Cakung   
49250      2                                   PHB Metropos KIP    Cakung   
49251      2                                 PHB Warung Jengkol    Cakung   

             Wilayah Panjang/Luas Satuan_Panjang/Luas  Tanggal  \
49247  Jakarta Timur          800                   m       28   
49248  Jakarta Timur          700                   m       28   
49249  Jakarta Timur         1000                   m       28   
49250  Jakarta Timur         1000                   m       28   
49251  Jakarta Timur          400                   m       28   

      Volume_Sampah_Perhari(M3)  
49247                       NaN  
49248                       NaN  
49249                       NaN  
49250                       NaN  
49251                       NaN

In [255]:
# Mengubah tipe data di kolom "Panjang/Luas" dari string menjadi float
data_februari[["Panjang/Luas"]] = data_februari[["Panjang/Luas"]].apply(pd.to_numeric)

In [256]:
# Mengubah tipe data pada kolom "Volume_Sampah_Perhari(M3)" dari string menjadi float
data_februari["Volume_Sampah_Perhari(M3)"] = pd.to_numeric(data_februari["Volume_Sampah_Perhari(M3)"], errors="coerce")

data_februari.dtypes

Bulan                          int64
Titik_Lokasi                  object
Kecamatan                     object
Wilayah                       object
Panjang/Luas                 float64
Satuan_Panjang/Luas           object
Tanggal                        int64
Volume_Sampah_Perhari(M3)    float64
dtype: object

Selain kolom "Panjang/Luas" dan kolom "Volume_Sampah_Perhari(M3)", saya juga akan mengubah format tipe data dari kolom "Satuan_Panjang/Luas". Walaupun tidak termasuk ke dalam kolom untuk memprediksi volume sampah yang berhasil diangkut di setiap harinya, tetapi kolom "Panjang/Luas" dan "Satuan_Panjang/Luas" akan saya buat visualisasinya. Maka dari itu kedua kolom ini tidak boleh memiliki nilai kosong (NaN) karena ditakutkan akan terjadi kesalahpahaman pembaca dalam menafsirkan kesimpulan dari grafik yang saya buat.

In [257]:
# Melihat nilai total dari masing-masing data yang ada di kolom "Satuan_Panjang/Luas"
data_februari["Satuan_Panjang/Luas"].value_counts()

m     26964
M      1428
Ha      784
H        84
HA       56
M³       28
km       28
m²       28
m2       28
Name: Satuan_Panjang/Luas, dtype: int64

Berdasarkan hasil diatas dapat dilihat bahwa nilai data hektar (yang dilambangkan dengan Ha) memiliki banyak sekali penulisan yang berbeda-beda seperti H dan HA padahal semuanya memiliki satu kesatuan yaitu Ha. Maka dari itu saya akan menjadikan nilai data H, dan HA ini menjadi satu kesatuan saja yakni data Ha.

Selain itu nilai dari data seperti M akan saya jadikan sebagai nilai m saja dan nilai m2 juga akan saya gabungkan dengan nilai m².

In [258]:
# Mengganti nilai data yang mirip seperti Ha, m, dan m²

# Mengganti nilai "H" dan "HA"
data_februari = data_februari.replace(to_replace =["H", "HA"], value="Ha")

# Mengganti nilai "M" menjadi "m"
data_februari = data_februari.replace(to_replace = "M", value="m")

# Mengganti nilai "m2" menjadi "m²"
data_februari = data_februari.replace(to_replace = "m2", value="m²")

data_februari["Satuan_Panjang/Luas"].value_counts()

m     28392
Ha      924
m²       56
M³       28
km       28
Name: Satuan_Panjang/Luas, dtype: int64

In [259]:
# Mengubah tipe dan nilai data pada kolom "Satuan_Panjang/Luas" dari yang semula string menjadi integer
from sklearn.preprocessing import LabelEncoder

label_encoder = LabelEncoder()

data_februari["Satuan_Panjang/Luas"] = label_encoder.fit_transform(data_februari["Satuan_Panjang/Luas"])

data_februari.head()

Bulan                                 Titik_Lokasi   Kecamatan  \
0      2       Kali / Sungai Sentiong - Ruas Dewi Sri  Johar Baru   
1      2         Kali / Sungai Sentiong - Ruas Abapon  Johar Baru   
2      2     Kali / Sungai Sentiong - JL Tanah Tinggi  Johar Baru   
3      2  Kali / Sungai Sentiong Jl.Galur - Pombensin  Johar Baru   
4      2           Kali / Sungai Sentiong - Jl. Timah  Johar Baru   

         Wilayah  Panjang/Luas  Satuan_Panjang/Luas  Tanggal  \
0  Jakarta Pusat         464.0                    3        1   
1  Jakarta Pusat         464.0                    3        1   
2  Jakarta Pusat         464.0                    3        1   
3  Jakarta Pusat         606.0                    3        1   
4  Jakarta Pusat         310.0                    3        1   

   Volume_Sampah_Perhari(M3)  
0                        2.0  
1                        3.0  
2                        3.0  
3                        4.0  
4                        7.0

In [260]:
data_februari.describe()

Bulan   Panjang/Luas  Satuan_Panjang/Luas       Tanggal  \
count  49252.0   39788.000000         49252.000000  49252.000000   
mean       2.0    3094.522871             3.748152     14.500000   
std        0.0   16862.952089             1.105949      8.077829   
min        2.0       0.000000             0.000000      1.000000   
25%        2.0     500.000000             3.000000      7.750000   
50%        2.0     800.000000             3.000000     14.500000   
75%        2.0    1400.000000             5.000000     21.250000   
max        2.0  350070.000000             5.000000     28.000000   

       Volume_Sampah_Perhari(M3)  
count               33619.000000  
mean                    2.588156  
std                     5.965463  
min                     0.000000  
25%                     0.000000  
50%                     0.500000  
75%                     2.000000  
max                   312.000000

In [261]:
# Mengecek kembali nilai NaN di masing-masing kolom yang ada di dataframe
data_februari.isnull().sum()

Bulan                            0
Titik_Lokasi                     0
Kecamatan                        0
Wilayah                          0
Panjang/Luas                  9464
Satuan_Panjang/Luas              0
Tanggal                          0
Volume_Sampah_Perhari(M3)    15633
dtype: int64

In [262]:
# Mengganti nilai NaN yang ada disetiap kolom di dataframe data_februari dengan nilai mean
mean = data_februari.mean()
data_februari = data_februari.replace(np.nan, mean)

data_februari.isnull().sum()

Bulan                        0
Titik_Lokasi                 0
Kecamatan                    0
Wilayah                      0
Panjang/Luas                 0
Satuan_Panjang/Luas          0
Tanggal                      0
Volume_Sampah_Perhari(M3)    0
dtype: int64

Setelah mengisi nilai kosong (NaN) dengan nilai mean (rata-rata) selanjutnya saya akan mengubah tipe data pada kolom "Panjang/Luas" dan "Volume_Sampah_Perhari(M3)" dari yang semula float menjadi integer.

In [263]:
# Mengubah tipe data pada kolom "Panjang/Luas" dan "Volume_Sampah_Perhari(M3)" dari float menjadi integer
data_februari["Panjang/Luas"] = data_februari["Panjang/Luas"].astype(int)
data_februari["Volume_Sampah_Perhari(M3)"] = data_februari["Volume_Sampah_Perhari(M3)"].astype(int)

data_februari.tail()

Bulan                                       Titik_Lokasi Kecamatan  \
49247      2  Kali Cakung Drain (Jembatan Tinggi-Perbatasan ...    Cakung   
49248      2                                   PHB Rawa Teratai    Cakung   
49249      2                                 PHB Rawa Gelam III    Cakung   
49250      2                                   PHB Metropos KIP    Cakung   
49251      2                                 PHB Warung Jengkol    Cakung   

             Wilayah  Panjang/Luas  Satuan_Panjang/Luas  Tanggal  \
49247  Jakarta Timur           800                    3       28   
49248  Jakarta Timur           700                    3       28   
49249  Jakarta Timur          1000                    3       28   
49250  Jakarta Timur          1000                    3       28   
49251  Jakarta Timur           400                    3       28   

       Volume_Sampah_Perhari(M3)  
49247                          2  
49248                          2  
49249                          2  
49250                          2  
49251                          2

In [264]:
# Menghapus kolom yang sudah tidak dibutuhkan yaitu kolom "Bulan"
data_februari = data_februari.drop(["Bulan"], axis=1)

In [265]:
# Mengecek nilai outliers untuk kolom "Panjang/Luas" dengan nilai quartil dan IQR
Q1 = data_februari["Panjang/Luas"].quantile(0.25)
Q3 = data_februari["Panjang/Luas"].quantile(0.75)

# Menghitung interquartile range maka rumusnya adalah quartil 3 (75%) - quartil 1 (25%)
IQR = Q3 - Q1

# Melihat ukuran (baris dan kolom) sebelum data yang outliers dibuang
print("Ukuran Data dari Kolom Panjang/Luas Sebelum Nilai Outliers Dibuang: ", data_februari.shape)

# Menghapus nilai outlier dari kolom "Panjang/Luas"
data_februari = data_februari[~((data_februari["Panjang/Luas"] < (Q1 - 1.5 * IQR)) | 
                              (data_februari["Panjang/Luas"] > (Q3 + 1.5 * IQR)))]

# Melihat ukuran (baris dan kolom) setelah data outliers dibuang dari kolom "Panjang/Luas"
print("Ukuran Data dari Kolom Panjang/Luas Setelah Nilai Outliers Dibuang: ", data_februari.shape)

Ukuran Data dari Kolom Panjang/Luas Sebelum Nilai Outliers Dibuang:  (49252, 7)
Ukuran Data dari Kolom Panjang/Luas Setelah Nilai Outliers Dibuang:  (47656, 7)


Terdapat 1.596 nilai outliers yang ada di kolom "Panjang/Luas". Hal ini sangat jauh berbeda jika dibandingkan dengan data di bulan Januari 2021 yang memiliki nilai outliers yang sangat banyak yakni berjumlah 10.540 data di kolom yang sama (Kolom "Panjang/Luas"). 

## 3. Data Visualisasi + Analisa Data di Grafiknya